Source/RuntimeImageLoader/Private/Helpers/GIFLoader.cpp

```diff
@@ -0,0 +1,220 @@
+// Copyright 2023 Peter Leontev and Muhammad Ahmed Saleem. All Rights Reserved.
+
+#include "GIFLoader.h"
+#include "RuntimeImageLoaderLog.h"
+#include "RuntimeImageLoaderLibHandler.h"
+
+DEFINE_LOG_CATEGORY(LibNsgifHelper);
+
+#if WITH_LIBNSGIF
+void* FRuntimeGIFLoaderHelper::bitmap_create(int width, int height)
+{
+	if (width > 4096 || height > 4096) {
+		return NULL;
+	}
+
+	return calloc(width * height, BYTES_PER_PIXEL);
+}
+
+unsigned char* FRuntimeGIFLoaderHelper::bitmap_get_buffer(void* bitmap)
+{
+	return (unsigned char*)bitmap;
+}
+
+void FRuntimeGIFLoaderHelper::bitmap_destroy(void* bitmap)
+{
+	free(bitmap);
+}
+
+uint8_t* FRuntimeGIFLoaderHelper::Load_File(const char* path, size_t* data_size)
```

Comment:
Yeah However, I just ain't moved with that at very first place because there would be an overhead again to parse the data of array. Prefer Raw Pointer uint8* over array.

Reply:
There will be literally no overhead as TArray<uint8> is a very basic wrapper around uint8* !

When you do `const TArray<uint8>& GifData = FFileHelper::LoadFileToArray(const char* path);` you can then get uint8* by using the following method: `GifData.GetData()`

Additionally, we don't need unsafe C functions (fopen, stat, fprintf, fread, fclose) in Unreal C++. You need to change this.